### ISSUE TITLE
Makefile:153: *** target pattern contains no '%'.  Stop.

### ISSUE THREAD

**Issue by lruzicka** [asb2m10/dexed#284]
Hello,
I cannot compile on Linux because I am getting the above error message. Would you please help me solve it? I tried Fedora 34 and Ubuntu 20 LTS, but I am getting the same result.

**Comment by ericksonsv** [comment]
Hey any answer from developers?

**Comment by lruzicka** [comment]
> Hey any answer from developers? same problem here!...

Unfortunately not. :/

**Comment by asb2m10** [comment]
This is very generic error, unless I have access to the build machine, it is difficult to pinpoint.

Do a `make -n` and see if the filenames that are referenced contains an special character in the filename. Make is very picky in filenames and if it find a name that can be part of a regular expression, it will refuse to continue.

Otherwise, post the output of the `make -n`

**Comment by monarrk** [comment]
Hi, I'm getting this error on openSUSE. this thread looks like it's pretty dead but I thought I'd try to give some more information to see if you can fix it, since I don't know anything about using JUCE.

The error comes from line 153 of the generated Makefile you get from projucer. That line looks like this:
```
VST3 : $(JUCE_OUTDIR)/$(JUCE_TARGET_VST3)
```

In vim, when I open it, it also has a `^M` char at the end of the line and the line after it, which is unusual. I don't know whether this is supposed to be a template or if it isn't generating the body of the build instructions. Does any of this sound fixable to you?

**Comment by lruzicka** [comment]
I can confirm the `^M` however when I removed it, it did not have any impact on the result which was still the same. When I have some more time, I will try to build it again on my machine and see if I can get any more details about the error.

**Comment by cwalloth** [comment]
same issue here, Chromebook Linux environment (Ubuntu)

**Comment by cwalloth** [comment]
$ make -n Makefile
Makefile:153: *** target pattern contains no '%'.  Stop.

> Do a `make -n` and see if the filenames that are referenced contains an special character in the filename. Make is very picky in filenames and if it find a name that can be part of a regular expression, it will refuse to continue.
> 
> Otherwise, post the output of the `make -n`

**Comment by twerq** [comment]
> The error comes from line 153 of the generated Makefile you get from projucer. That line looks like this:
> 
> ```
> VST3 : $(JUCE_OUTDIR)/$(JUCE_TARGET_VST3)
> ```

### tldr:
You should install C++ compiler. I use debian based distro and install C++ like this:

```
sudo apt install g++
``` 

### Explanation
Varialble `JUCE_TARGET_VST3` to be populated involves another variable `CXX`. And variable `CXX` populated when C++ path exists. Otherwise  `JUCE_TARGET_VST3` remains empty and build script throws `target pattern contains no '%'`.

**Comment by asb2m10** [comment]
Thanks for the explanation.
